modules/mbedtls/configs/config-tls-generic.h

```diff
@@ -28,6 +28,11 @@
 #define MBEDTLS_HAVE_ASM
 #endif
 
+#if defined(CONFIG_MBEDTLS_LMS)
+#define MBEDTLS_LMS_C
+#define PSA_WANT_ALG_SHA_256 1
```

Comment:
You don't need to define the symbol to 1. Follow the pattern and just `#define PSA_WANT_ALG_SHA_256`

Reply:
In my test the compiler complained if I do that since it conflicted with a later definition of PSA_WANT_ALG_SHA256 1 in config_psa.h:

```
mbedtls/include/mbedtls/config_psa.h:697: error: "PSA_WANT_ALG_SHA_256" redefined [-Werror]
  697 | #define PSA_WANT_ALG_SHA_256 1
  [...]
  
mbedtls/configs/config-tls-generic.h:33: note: this is the location of the previous definition
   33 | #define PSA_WANT_ALG_SHA_256
  ```

Reply:
All new configurations in Mbed TLS that follows the pattern PSA_WANT_ALG_XXXX uses 1 as value.
Only older configurations use naked defines. 

There will be redefined-warnings if we don't follow the established patterns (for both new and old)

Reply:
The lms source code is a bit of an odd-ball in terms of crypto feature. It is written with legacy prefixed APIs (mbedtls_) because there exists no defined PSA Crypto API for the given group of algorithms. We have raised the question whether or not this will be supported in PSA Crypto APIs in due time, but have gotten no definite answer on this.

The configuration for the LMS feature is using "legacy style" configuration: MBEDTLS_LMS_C while [the code](https://github.com/Mbed-TLS/mbedtls/blob/development/library/lms.c#L100) is asking for PSA Crypto API features

If this feature is deemed important for Zephyr and should go in, then you have to accept intermix between new and old, unfortunately...

Reply:
@joerchan gentle ping?

Reply:
@mfischer I'm not sure what you are pinging for? I have approved the PR.

If you waiting fro merging, then ping @d3zd3z :)